### DISCUSSION TITLE
GitHub action: Use a variable inside other variable

### DISCUSSION THREAD

**Discussion by hocinemala** [community/community#23473] [New to GitHub]
<p>Hello,</p>
<p>In a GitHub action, how can I set an env variable with a variable inside other variable like this:</p>
<pre><code class="lang-auto">env:
    TF_CLIENT_ID: '${{ tf-env-${{ inputs.env_id }} }}'
</code></pre>
<p>Regards</p>

**Comment by airtower-luna** [comment]
<p>Use the <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#format"><code>format()</code> function</a>, something like this should work:</p>
<pre><code class="lang-yaml">env:
  TF_CLIENT_ID: '${{ format('tf-env-{0}', inputs.env_id) }}'
</code></pre>

**Comment by hocinemala** [comment]
<aside class="quote group-Advisory-Council" data-username="airtower-luna" data-post="2" data-topic="253116">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p><code>${{ </code></p>
</blockquote>
</aside>
<p>The result is ‘tf-env-d’ if  “inputs.env_id = d”<br>
but I want the value of the variable ‘tf-env-d’ like ${{ tf-env-d }}</p>
<p>this is not working: ${{ ${{ format(‘tf-env-{0}’, inputs.env_id) }} }}</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="hocinemala" data-post="3" data-topic="253116">
<div class="title">
<div class="quote-controls"></div>
 hocinemala:</div>
<blockquote>
<p>but I want the value of the variable ‘tf-env-d’ like ${{ tf-env-d }}</p>
</blockquote>
</aside>
<p>Ah, that’s a very different thing. You can use <code>format()</code> to build the index for a mapping, like so:</p>
<pre><code class="lang-auto">${{ env[format(...)] }}
</code></pre>

**Reply by Gufderald** [reply]
Man, you are the savior. Thank you! Bad of github that this feature is not fully documented.

**Reply by AnimMouse** [reply]
```
${{ env[format('tf-env-{0}', inputs.env_id)] }}
```

**Comment by hocinemala** [comment]
<p>it works fine thank you</p>
